src/client.js

```diff
@@ -5,23 +5,51 @@ const url = require("url");
 
 // Client can read/write messages from a TCP server
 class Client {
-  // init initializes the Client
-  init(addr) {
-    let u = url.parse("tcp://" + addr, false, false);
-    this.socket = new net.Socket();
-    this.socket.connect(u.port, u.hostname, function() {});
-    this.socket.on("close", function() {
-      process.exit();
-    });
-    return this;
-  }
-
-  // write writes an event to the server
-  write(targetID, eventName, payload) {
-    let data = { name: eventName, targetID: targetID };
-    if (typeof payload !== "undefined") Object.assign(data, payload);
-    this.socket.write(JSON.stringify(data) + "\n");
-  }
+    // init initializes the Client
+    init(addr) {
+
+        this.connect(addr)
+
+        this.socket.on('error', function(err){
+            // Writing to a file in case of error related to socket
+            var fs = require('fs');
+            fs.appendFile("/tmp/astilectron.log", "Socket Error: "+err, function(e){
+                console.log("Error while writing to file:"+e);
+            })
+            process.exit()
+        })
+
+        this.socket.on('close', function() {
+            process.exit()
+        })
+        return this
+    }
+
+    // write writes an event to the server
+    write(targetID, eventName, payload) {
+        let data = {name: eventName, targetID: targetID}
+        if (typeof payload !== "undefined") Object.assign(data, payload)
+        this.socket.write(JSON.stringify(data) + "\n")
+    }
+
+    /*
+     * for proper socket closing, unix socket remains open even after
+     * quiting the application, this function ends the socket connection
+     */
+    close() {
+        this.socket.end()
+    }
+
+    // establishes connection based on underlying OS
+    connect(addr) {
+        if ( os.platform() != "win32" ) {
```

Comment:
`addr` here should be the full `addr` hence `tcp://127.0.0.1:4003` or `unix://path/to/socket`.

That way, instead of creating the connection based on OS, it should be created based on the `addr` scheme (`tcp://` or `unix://`)

The `go` side will have to be updated as well.

Reply:
Going with this approach, we first have to check for "tcp://" or "unix://" and then split the string to get the "host: port" or "path" and then pass it to either socket.connect() or net.createConnection().

If we use net.createConnection(), then the possible arguments to this function are:
1. {port: 1234}
2. {path: '/tmp/xyz.sock'}

So my confusion is why don't we go with the first approach, as there are fewer checks to be done and the code/logic seems simpler.

@asticode it would be helpful if you can explain it with some details?

Reply:
My problem with the first approach is that both GO and JS have the responsibility of determining the connection scheme based on the OS. As far as I'm concerned, that shouldn't be the case since only one side should have this logic. Otherwise, when we update such a logic, we'll have to update both GO and JS.

Hence I'd rather have GO driving the value (and scheme) of `addr` based on the OS and JS creating the proper connection based on the scheme of the received `addr`. That way reponsibilities are properly divided.

In a nutshell:

```
GO decides the value of addr based on the OS
|
GO sends the value of addr to JS through a flag
|
JS creates the proper connection based on the addr (and scheme) received
```

That way, if we decide to update the `addr` logic, we only have to update GO.

Does that make more sense?

Reply:
Yes, understood your point of view, I have made those changes as per your suggestions.
Testing those changes, hopefully I will push the changes soon.